### ISSUE TITLE
Easy way to omit parameter when using ?

### ISSUE THREAD

**Issue by frankoid** [lemonlabsuk/scala-uri#16]
I found the 0.5.7 behaviour of omitting the query parameter altogether if the value was `None` very useful.

Code Example 0.5.7:
```scala
scala> import com.netaporter.uri.dsl._
import com.netaporter.uri.dsl._

scala> val optional: Option[String] = None
optional: Option[String] = None

scala> "foo" ? ("optional" -> optional)
res0: com.netaporter.uri.Uri = foo
```

Would it be possible to revert to the 0.5.7 behaviour or add it as an alternative?

With 1.0.0 it seems I'd have to write the following ugly code to omit the parameter when the value is `None`:

```scala
val optional: Option[String] = ???
val url = optional.fold("http://example.com/foo" ? ("bar" -> bar))(_ => "http://example.com/foo" ? ("bar" -> bar) & ("optional" -> optional)))
```

Play interprets a query parameter without a value as having a value of the empty string, and the URL I'm generating is being passed to a Play app.

When I discovered that scala-uri omitted the parameter altogether when the value was `None` it was a real 😍 moment so I'd love to have this feature back.

**Comment by theon** [comment]
Absolutely. I will a new methods to do this.

0.5.x had a bunch of `addParam` methods dotted about that behaved inconsistently; some only added  params with `Some`, others added both `Some` and `None`s. Now there is consistency I'll add new methods to do this.

**Comment by theon** [comment]
I had trouble coming up with good method names (especially for the DSL), so I opted for adding this as a config option.

```scala
import io.lemonlabs.uri.config._
import io.lemonlabs.uri.dsl._
implicit val config: UriConfig = UriConfig(renderQuery = ExcludeNones)

"foo" ? ("optional" -> None)
res1: io.lemonlabs.uri.Url = foo
```
This is released as version `1.1.1`

Hope it fits your needs. If not, let me know.

**Comment by frankoid** [comment]
Wow! That's great, thank you, now happily on `1.1.1`.
